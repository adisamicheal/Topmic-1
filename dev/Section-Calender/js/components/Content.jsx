import React from 'react';
import Arrow from './Arrow';
import Event from './Event';
export default class Content extends React.Component{
	constructor(){
		super();
	}
	render(){
		return(
			<div className="heroImage">
				<div className="line"></div>
				<h2><strong>Up Coming Events</strong></h2>
				<div className="renderEvents">
						<Event {...this.props}/>
				</div>
				<Arrow
				{...this.props}
				 />
			</div>
		)
	}
}
