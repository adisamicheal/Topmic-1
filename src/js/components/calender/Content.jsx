import { number } from 'prop-types';
import React from 'react';
import Arrow from './Arrow';
import Event from './Event';

const Content = ({ count }) => (
	<div className="heroImage">
		<div className="line" />
		<h2>
			<strong>Up Coming Events</strong>
		</h2>
		<div className="renderEvents">
			<Event count={count} />
		</div>
		<div className="clear" />
		<Arrow />
	</div>
);
Content.propTypes = {
	count: number.isRequired
};
export default Content;
