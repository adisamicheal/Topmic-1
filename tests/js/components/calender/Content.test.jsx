import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Content from '../../../../src/js/components/calender/Content';

Enzyme.configure({ adapter: new Adapter() });

describe('Test the Content Component of Calender', () => {
	const count = 0;
	const wrapper = shallow(<Content count={count} />);
	it('checks the rendering', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
