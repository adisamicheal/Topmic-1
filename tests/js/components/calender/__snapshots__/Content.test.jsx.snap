// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Content Component of Calender checks the rendering 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Content
    count={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="line"
        />,
        <h2>
          <strong>
            Up Coming Events
          </strong>
        </h2>,
        <div
          className="renderEvents"
        >
          <Connect(Event)
            count={0}
          />
        </div>,
        <div
          className="clear"
        />,
        <Connect(Arrow) />,
      ],
      "className": "heroImage",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "line",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <strong>
            Up Coming Events
          </strong>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Up Coming Events",
          },
          "ref": null,
          "rendered": "Up Coming Events",
          "type": "strong",
        },
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Event)
            count={0}
          />,
          "className": "renderEvents",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "count": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "clear",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="line"
          />,
          <h2>
            <strong>
              Up Coming Events
            </strong>
          </h2>,
          <div
            className="renderEvents"
          >
            <Connect(Event)
              count={0}
            />
          </div>,
          <div
            className="clear"
          />,
          <Connect(Arrow) />,
        ],
        "className": "heroImage",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "line",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <strong>
              Up Coming Events
            </strong>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Up Coming Events",
            },
            "ref": null,
            "rendered": "Up Coming Events",
            "type": "strong",
          },
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Event)
              count={0}
            />,
            "className": "renderEvents",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "count": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "clear",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
